asserts/database.go

```diff
@@ -89,6 +89,18 @@ var (
 	ErrNotFound = errors.New("assertion not found")
 )
 
+// SupersededRevisionError is the error returned when an assertion is added
```

Comment:
Shortening to SupersededError seems fine to me, but "no longer current" isn't quite accurate - the error is also returned when the currently stored revision is the *same* as the revision of the assertion being added.

Making Current public could be useful, but is not essential to the use-case I needed an error type for. If I knew the current revision was different to the revision being added I could skip one database lookup.